rosetta/docs/GPU_performance.md

```diff
@@ -0,0 +1,131 @@
+    - --xla_gpu_all_reduce_combine_threshold_bytes=8589934592
+
+
+### Flags for Collective permute 
+
+The following flags enable overlap of pipeline parallel communication of send/recv with computation. 
+- --xla_gpu_enable_pipelined_p2p=true  (false by default)
+- --xla_gpu_collective_permute_decomposer_threshold=33521664
+
+### Flags to enable Collective Matmul
+
+The following flags enable overlap of tensor parallel communication with GEMMs/matmul by splicing GEMMs into smaller chunks and triggering each chunks' collective right after the chunk's GEMM is done. The threshold determines the size of output buffer of GEMM when this optimization becomes active (0 enables collective matmul for all GEMM-collective patterns)
+- --xla_gpu_multi_streamed_windowed_einsum=true
+- --xla_gpu_threshold_for_windowed_einsum_mib=0
+
+### Profile Guided Latency Estimator (PGLE)
+
+The following flag enables use of PGLE with JAX/XLA. Please see [PGLE notes](https://github.com/NVIDIA/JAX-Toolbox/blob/main/rosetta/docs/PGLE.md) for more details.
+- --xla_gpu_pgle_profile_file_or_directory_path=filename
+
+## Other XLA Flags 
+
+### Enable CUDA graphs
+
+The below enables CUDA Graph suppport for JAX/XLA workloads
+- --xla_gpu_enable_command_buffer (1 by default, available options are 0, 1, 2, and 3)
+
+### Dynamic-Update Slice Fusion
+
+The following flag removes extra copies introduced by DUS (dynamic update slice) when used in conjunction with custom NVIDIA kernels (like cuBLAS for GEMMs)
+- --xla_gpu_enable_custom_fusions=true
+
+### NCCL Optimizations
+
+Enable user-buffers in NCCL for zero-copy collectives and send/recv. Needs NCCL_NVLS_ENABLE=1 for AG, AR, RS.
+- --xla_gpu_enable_nccl_user_buffers=true
+
+Flags to improves memory consumed by NCCL.
+- --xla_gpu_enable_nccl_resource_sharing=true  
+- --xla_gpu_enable_nccl_per_stream_comms=false [https://github.com/openxla/xla/pull/9845](https://github.com/openxla/xla/pull/9845)
+
+Fine-grain control to improve performance by initializing a NCCL communicators to use only max_nchannels (SMs) 
+- --xla_gpu_nccl_collective_max_nchannels
+- --xla_gpu_nccl_p2p_max_nchannels
+
+### Debug flags 
+- --xla_dump_to=some/path
+- --xla_dump_latency_hiding_schedule=true
+- --xla_gpu_lhs_enable_gpu_async_tracker=false
+
+### Miscellaneous flags 
+- --xla_gpu_enable_triton_softmax_fusion=false (disable triton softmax if using a custom kernel for MHA for example)
+- --xla_gpu_cudnn_gemm_fusion=true (enables GEMM/bias fusion via cuDNN)
+- --xla_gpu_enable_cudnn_fmha=false (enables XLA pattern matcher to detect multi-headed attention pattern in JAX)
```

Comment:
Why set it to false? Normally true should be faster and is not the default?

Reply:
I set it to false when using TE.

Reply:
I guess it conflicts with TE/cudnn kernel for MHA ?

Reply:
I updated to "--xla_gpu_enable_triton_softmax_fusion=false (disable if using a custom kernel for MHA; leave to default/true otherwise)". I think this is fine.

Reply:
I don't understand why you need to set that flag when using TE.
When using TE, the Pattern Matcher won't find anything, so it will do nothing.

I'm good with the current descriptions. This is a Miscellaneous flag.
But can you remove it from the flags you try? If you need to use it, make a bug report as it isn't normal.

Reply:
Removed.